install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -233,13 +239,53 @@ spec:
         {{- end }}
         {{- end }}
         securityContext:
+          {{- if .Values.securityContext.privileged }}
           privileged: true
+          {{- else }}
```

Comment:
if the `privileged` flag is not set, you can disable make the container "unconfined" for SELinux (which would be needed here), by setting the following:

```
securityContext:
  seLinuxOptions:
    level: 's0'
    type: 'spc_t'
```

The level might not have been strictly needed here, but it's good to add just in case.